op-e2e/faultproofs/output_cannon_test.go

```diff
@@ -307,6 +305,7 @@ func TestOutputCannonStepWithKZGPointEvaluation(t *testing.T) {
 		receipt := sendKZGPointEvaluationTx(t, sys, "sequencer", sys.Cfg.Secrets.Alice)
 		precompileBlock := receipt.BlockNumber
 		t.Logf("KZG Point Evaluation block number: %d", precompileBlock)
+		require.NoError(t, wait.ForSafeBlock(ctx, sys.RollupClient("sequencer"), precompileBlock.Uint64()))
```

Comment:
This is done automatically in `StartOutputCannonGame`: https://github.com/ethereum-optimism/optimism/blob/03759658e4a860852c7d676b4f00fe5564d9ca89/op-e2e/e2eutils/disputegame/helper.go#L231

Reply:
That's surprising. One reason the test flakes is because cannon can't find a step that uses the precompile. So I assumed this was because the safe head didn't include it.
I'll run this a couple times without this line on CI infra to see if it still flakes.

Reply:
The waitForSafeHead is replaced with a [wait](https://github.com/ethereum-optimism/optimism/pull/9577/files#diff-a9f4b6a7a5985d2f6ac1b397b0c2fb8c2c3709b55274b1b08ad836bd3db6b6d6R315) to ensure that we include the precompile tx is included in the next epoch (note the dispute game system's seqWindowSize is 4).
I'm not totally sure why this works, but it's effective at eliminating flakes in CI. Derivation is a bit weird around genesis so I'm not worried about introducing this "hack".